languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/middlewares/CtpOAuthMiddlewareForClientCredentialsGrant.java

```diff
@@ -0,0 +1,79 @@
+package client.middlewares;
+
+import client.Middleware;
+import client.MiddlewareArg;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+import java.util.concurrent.CompletableFuture;
+
+public final class CtpOAuthMiddlewareForClientCredentialsGrant implements Middleware {
+
+    private final String clientId;
+    private final String clientSecret;
+    private final String accessTokenEndpoint;
+    private final String authorizationBaseUrl;
+    private final String scope;
+    private final OAuth20Service oAuth20Service;
+
+
+    public CtpOAuthMiddlewareForClientCredentialsGrant(String clientId, String clientSecret, String scope, String accessTokenEndpoint, String authorizationBaseUrl) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.accessTokenEndpoint = accessTokenEndpoint;
+        this.authorizationBaseUrl = authorizationBaseUrl;
+        this.scope = scope;
+        oAuth20Service = new ServiceBuilder(clientId)
+                .apiSecret(clientSecret)
+                .defaultScope(scope)
+                .build(new CtpApi2());
+    }
+
+
+    public String getClientId() {
+        return clientId;
+    }
+
+    public String getClientSecret() {
+        return clientSecret;
+    }
+
+    public String getAccessTokenEndpoint() {
+        return accessTokenEndpoint;
+    }
+
+    public String getAuthorizationBaseUrl() {
+        return authorizationBaseUrl;
+    }
+
+    public String getScope() {
+        return scope;
+    }
+
+    @Override
+    public CompletableFuture<MiddlewareArg> next(MiddlewareArg arg) {
+
+        try {
+            // We have to do caching here
```

Comment:
This is a `TODO`, right? If I'm right, I would add `TODO` to the comment.

Reply:
if this asks for a new access token every time we execute it, maybe we should really add caching before merging. Just running integration tests would require almost 150 tokens.

Reply:
@alex-ct i see your concern here, i would add caching here.

Reply:
I ll mark it as a Todo here

Reply:
Its just that it's not evident, i ll just mark it as a TODO